ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -581,6 +581,80 @@ public func catch<T, E, F>(handler: E -> SignalProducer<T, F>)(producer: SignalP
 	}
 }
 
+/// Describes how multiple signals or producers should be joined together.
+public enum JoinStrategy: Equatable {
+	/// The signals should be merged, so that any value received on any of the
+	/// input signals will be forwarded immediately to the output signal.
+	///
+	/// The resulting signal will complete only when all inputs have completed.
+	case Merge
```

Comment:
In RAC 2, `-flatten:` takes a max number of signals to flatten concurrently (https://github.com/ReactiveCocoa/ReactiveCocoa/blob/54098f8b457d5fa6b2a285d790c2c07c099b98f5/ReactiveCocoa/RACSignal%2BOperations.h#L221). Would we want to parameterize `.Merge` with that?